doc/sphinx/language/core/conversion.rst

```diff
@@ -3,11 +3,75 @@
 Conversion rules
 ----------------
 
-|CiC| has conversion rules that can be used to determine if two
-terms are equal by definition, or :term:`convertible`.
-Conversion rules consist of reduction rules and expansion rules.
-See :ref:`applyingconversionrules`,
-which describes tactics that apply these conversion rules.
+Conversion rules convert terms to other terms that are equivalent by
+definition under the rules of |CiC|.  Conversion rules can be used, for example,
+to determine if two terms are equal by definition, or :term:`convertible`, by
+converting both terms to a normal form, then verifying they are syntactically
+equal (ignoring differences in the names of bound variables
+:term:`alpha-convertibility <alpha-convertible>`).
+
+Conversion rules can be reductions or expansions.
+Reductions convert terms to something that is incrementally closer to its
+normal form.  For example, :term:`zeta-reduction` removes
+:n:`let @ident := @term__1 in @term__2` constructs from a term by replacing
+:n:`@ident` with :n:`@term__1` wherever :n:`@ident` appears in :n:`@term__2`.
+The resulting term may be longer or shorter than the original.
+
+.. coqtop:: all
+
+   Eval cbv zeta in let i := 1 in i + i.
+
+Expansions are reductions applied in the opposite direction,
+for example expanding `2 + 2` to `let i := 2 in i + i`.  While applying
+reductions (except for eta-reduction) gives a unique result , the associated
+expansion may not give a unique result.  For example, `2 + 2` could also be
+expanded to `let i := 2 in i + 2`.  Expansions that give a unique
+result are also referred to as *contractions*.
+
+The normal form is defined as the result of applying a particular
+set of conversion rules (beta-, delta-, iota- and zeta-reduction and eta-expansion)
+repeatedly until it's no longer possible to apply any of them.
+
+Some reductions can be considered computations, for example reducing `2*3+4` with
+`cbv beta delta iota` to `10`, which requires applying several reduction rules
+repeatedly.
+
+The useful conversion rules are:
+
+   .. list-table::
+      :header-rows: 1
+
+      * - Conversion name
+        - Description
+
+      * - beta-reduction
+        - eliminates `fun`
+
+      * - delta-reduction
+        - replaces a defined symbol with its definition
+
+      * - cofix-reduction
+        - replace a `cofix` symbol with its definition
```

Comment:
Not quite.  When a `cofix` is applied to arguments such that it's structural argument is headed by a constructor, the cofix reduction will replace the cofix with a fun (leaving a beta redex) and replace any recursive calls with the original `cofix` term